pkg/mimir/client/http_client.go

```
@@ -126,5 +128,42 @@ func createHTTPClient() (*http.Client, error) {
 		}
 		timeout = time.Duration(timeoutSeconds) * time.Second
 	}
-	return &http.Client{Timeout: timeout}, nil
+
+	tlsConfig := &tls.Config{}
+	httpClient := http.Client{
+		Timeout: timeout, 
+		Transport: &http.Transport{TLSClientConfig: tlsConfig},
+	}
+
+	if c.config.TLS.CAPath != "" {
+		certPool, err := x509.SystemCertPool()
+		if err != nil {
+			return nil, err
+		}
+
+		caCertPEM, err := os.ReadFile(c.config.TLS.CAPath)
+		if err != nil {
+			return nil, err
+		} 
+		
+		ok := certPool.AppendCertsFromPEM(caCertPEM)
+		if !ok {
+			return nil, err
```

Error here is nil, you should return a custom error.